### DISCUSSION TITLE
question

### DISCUSSION THREAD

**Discussion by ghost** [fastapi/fastapi#9426] [Questions]
### First Check

- [x] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
async def create_upload_file(
        param: TextData,
        # file: UploadFile | None = None
        custom_dict: Optional[UploadFile] = File(
            "None",
        )

):
```

### Description

I defined a route function in FastAPI that includes an optional file upload and a list of TextData items. When I pass only the TextData parameter, the request succeeds and returns the correct results. However, when I pass both the optional file upload and TextData parameter at the same time, the API returns a 422 Unprocessable Entity error with the detail message "custom_dict parameter must be an uploaded file". I have tried different request formats, including passing both the file and TextData parameters in a multipart/form-data request, or passing the TextData parameter in application/json format, but none of them solved the problem. I hope to find a way to successfully call my route function when both the optional file upload and TextData parameters are passed at the same time.

### Operating System

Linux

### FastAPI Version

0.95.1

### Python Version

python3.10.11

**Comment by Viicos** [comment]
https://github.com/tiangolo/fastapi/pull/9383 might be related. I'll try to continue working on it

**Comment by YuriiMotov** [comment]
Duplicate with answer: https://github.com/fastapi/fastapi/discussions/9484
